src/core-services/email/mutations/retryFailedEmail.js

```diff
@@ -0,0 +1,42 @@
+import Logger from "@reactioncommerce/logger";
+
+/**
+ * @name emails/retryFailed
+ * @summary Retry a failed or cancelled email job
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - mutation input
+ * @param {String} input.jobId - a sendEmail job ID
+ * @returns {Boolean} - returns true if job is successfully restarted
+ */
+export default async function retryFailed(context, input) {
+  const { checkPermissions, backgroundJobs } = context;
+  const { jobId, shopId } = input;
+  let emailJobId = jobId;
+
+  await checkPermissions(["owner", "admin", "reaction-email"], shopId);
+
+  Logger.debug(`emails/retryFailed - restarting email job "${jobId}"`);
+
+  // Get email job to retry
+  const job = await backgroundJobs.getJob(jobId);
+
+  // If this job was never completed, restart it and set it to "ready"
+  if (job._doc.status !== "completed") {
+    await job.restart();
+    await job.ready();
+  } else {
+    // Otherwise rerun the completed job
+    // `rerun` clones the job and returns the id.
+    // We'll set the new one to ready
+    emailJobId = job.rerun();
+  }
+
+  // Set the job status to ready to trigger the Jobs observer to trigger sendEmail
+  backgroundJobs.update({ _id: emailJobId }, {
```

Comment:
This should be updating the `Emails` collection. You can get it from context like so.

```
  const { checkPermissions, backgroundJobs, collections } = context;
  const { Emails } = collections;
```

Reply:
It was earlier used for `backgroundJobs` but thanks for the feedback. I have made the changes now.